gradunwarp/core/coeffs.py

```diff
@@ -78,137 +78,74 @@ def get_siemens_coef(cfile):
-                   'Alpha_y': ay,
-                   'Alpha_z': az,
-                   'Beta_x': bx,
-                   'Beta_y': by,
-                   'Beta_z': bz}
+    txt_var_map = create_txt_var_map(coef_array_sz)
 
     coef_file_parse(cfile, txt_var_map)
 
-    return Coeffs(ax, ay, az, bx, by, bz, R0_m)
+    return Coeffs(
+        txt_var_map['alpha_y'],
+        txt_var_map['alpha_z'],
+        txt_var_map['alpha_x'],
+        txt_var_map['beta_y'],
+        txt_var_map['beta_x'],
+        txt_var_map['beta_z'],
+        R0_m)
 
 def grad_file_parse(gfile, txt_var_map):
-    ''' a separate function because GE and Siemens .coef files
-    have similar structure
-
-    modifies txt_var_map in place
-    '''
-    gf = open(gfile, 'r')
-    line = next(gf)
-    # skip the comments
-    while not line.startswith('#*] END:'):
-        line = next(gf)
-
-    # get R0
-    line = next(gf)
-    line = next(gf)
-    line = next(gf)
-    R0_m = float(line.strip().split()[0])
-
-    # go to the data
-    line = next(gf)
-    line = next(gf)
-    line = next(gf)
-    line = next(gf)
-    line = next(gf)
-    line = next(gf)
-    line = next(gf)
-
     xmax = 0
     ymax = 0
-
-    while 1:
-        lindex =  line.find('(')
-        rindex =  line.find(')')
-        if lindex == -1 and rindex == -1:
-            break
-        arrindex = line[lindex+1:rindex]
-        xs, ys = arrindex.split(',')
-        x = int(xs) 
-        y = int(ys)
-        if x > xmax:
-            xmax = x
-        if y > ymax:
-            ymax = y
-        if line.find('A') != -1 and line.find('x') != -1:
-            txt_var_map['Alpha_x'][x,y] = float(line.split()[-2])
-        if line.find('A') != -1 and line.find('y') != -1:
-            txt_var_map['Alpha_y'][x,y] = float(line.split()[-2])
-        if line.find('A') != -1 and line.find('z') != -1:
-            txt_var_map['Alpha_z'][x,y] = float(line.split()[-2])
-        if line.find('B') != -1 and line.find('x') != -1:
-            txt_var_map['Beta_x'][x,y] = float(line.split()[-2])
-        if line.find('B') != -1 and line.find('y') != -1:
-            txt_var_map['Beta_y'][x,y] = float(line.split()[-2])
-        if line.find('B') != -1 and line.find('z') != -1:
-            txt_var_map['Beta_z'][x,y] = float(line.split()[-2])
-        try:
-            line = next(gf)
-        except StopIteration:
-            break
-
-    # just return R0_m but also txt_var_map is returned
+    with open(gfile, 'r') as gf:
+        for line in gf:
+            re_search = re.search("(?P<no>\d+)[\s]+(?P<aorb>[AB])[\s]*\(\s*(?P<x>\d+),\s*(?P<y>\d+)\)\s+(?P<spectrum>[\-]?\d+\.\d+)\s+(?P<axis>[xyz])", line)
+            if re_search:
+                re_res = re_search.groupdict()
+                alphabeta = 'alpha' if re_res['aorb'] == 'A' else 'beta'
+                x, y = int(re_res['x']),int(re_res['y'])
+                txt_var_map[f"{alphabeta}_{re_res['axis']}"][x, y] = float(re_res['spectrum'])
```

Comment:
Looks like python2 doesn't like `f"..."`.  Can this be easily written compatibly?

Reply:
Oh! I didn't realize that it still aim at supporting python2. Will fix that!